### PR TITLE
Capability to add a default Pod TopologySpreadConstraints

### PR DESCRIPTION
**Pull request by a7i**
**Note**: Samson is a public repo, do not include Zendesk-internal information, urls, etc.

* Add description
* Add screenshots when changing the UI
* Add unit tests

### References
- Jira link: 

### Risks
- Low/Med/High: Low, it's guarded by an env var

### CODE REVIEW

File: plugins/kubernetes/README.md [hunk 1]
```diff
@@ -221,6 +221,20 @@ and increase the `spec.terminationGracePeriodSeconds` if necessary.
...
 
+### Adding a default Pod TopologySpreadConstraint
+
+In order to control how Pods are spread across the cluster among failure domains, you can set a default 
+[`TopologySpreadConstraint`](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/)
+to help achieve high availability as well as efficient resource utilization.
+
+(will only add if pod `metadata.annotations.samson/podTopologySpread` is not set to `disabled` and a `topologySpreadConstraints` is not defined)
+
+- Set `KUBERNETES_ADD_POD_TOPOLOGY_SPREAD` to enable
```
Comment (grosser) [inline plugins/kubernetes/README.md:232]: can we do this instead ?
```suggestion
- Set `KUBERNETES_POD_TOPOLOGY_SPREAD={config: here}` to enable
```

**Review by grosser — APPROVED** [review]
maybe the initial version could do `if annotation == enable` so we can test it out with a few deployments ?
... idk if you got local kubernetes setup running but it can be hard

**Review by uthark — APPROVED** [review]
I'd rather do it in kube-admission.

**Comment by jklaw90** [comment]
@uthark @grosser https://github.com/zendesk/cicd-toolkit/pull/1232 i did it in cicd too, should i remove it?

**Comment by grosser** [comment]
maybe best to have an rfc for this, can steal pro/cons from the slack thread and then we also have a nice "why" for whatever way we decide on

**Comment by a7i** [comment]
Going to close this PR since it can be achieved via admission webhook in a generic fashion rather than adding logic/complexity to Samson.

**Pull request closed without merging**